subprojects/native/src/main/java/org/gradle/internal/nativeintegration/filesystem/jdk7/NioFileMetadataAccessor.java

```diff
@@ -50,7 +51,7 @@ public FileMetadata stat(File file) {
             return DefaultFileMetadata.directory(accessType);
         }
         if (attributes.isOther()) {
-            return DefaultFileMetadata.missing(accessType);
+            throw new UncheckedIOException("Unsupported file type for " + file.getAbsolutePath());
```

Comment:
Can we log something here about the actual type?

Reply:
The type is "Other", so it doesn't make much sense to log it. I used to log it first and then realized that it doesn't make much sense.